### PR TITLE
refactor(arch): split rpc module to core types and utility

### PR DESCRIPTION
**Pull request by KaoImin**
**What this PR does / why we need it**:
Refactor the mercury architecture.

**Which issue(s) this PR fixes**:

Fixes #

**Which docs this PR relation**:

Ref #

**Which toolchain this PR adaption**:

No Breaking Change

**Special notes for your reviewer**:

### CODE REVIEW

File: core/rpc/core/src/impl/query.rs [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 
 use common::utils::parse_address;
 use common::{Context, Order, PaginationRequest, PaginationResponse, Range, SECP256K1};
 use common_logger::tracing_async;
-use core_storage::{DBInfo, Storage};
-use protocol::TransactionWrapper;
+use core_ckb_client::CkbRpc;
+use core_rpc_types::lazy::{CURRENT_BLOCK_NUMBER, CURRENT_EPOCH_NUMBER};
+use core_rpc_types::{
+    indexer, AddressOrLockHash, AssetInfo, Balance, BlockInfo, BurnInfo, GetBalancePayload,
```
Comment (rev-chaos) [inline core/rpc/core/src/impl/query.rs:10]: You should rebase the latest version.

Comment (KaoImin) [reply core/rpc/core/src/impl/query.rs:10]: The current code has already rebase the `dev` branch.

File: core/rpc/core/Cargo.toml [hunk 2]
```diff
@@ -1,15 +1,12 @@
 [package]
 name = "core-rpc"
-version = "0.2.0-beta.4"
-authors = ["Nervos Network"]
+version = "0.1.0"
```
Comment (rev-chaos) [inline core/rpc/core/Cargo.toml:3]: Why 0.1.0?

File: README.md [hunk 3]
```diff
@@ -135,11 +135,13 @@ The `main` branch is built and tested regularly, considered as production-ready;
...
+The `Mercury` crate's minimum supported rust version is 1.55.1.
 
 ---
 
 ## Documentations
 
 - [Mercury API Documentation](core/rpc/README.md)
+- [Mercury Architecture Documentation](docs/architecture.md)
 - [Mercury Config Documentation](docs/config.md)
+- [Mercury Layout Documentation](docs/layout.md)
```
Comment (rev-chaos) [inline README.md:147]: Should we append contribution doc here?

Comment (KaoImin) [reply README.md:147]: In my opinion, the documents listed here are related to mercury, and a should be placed in a separate section, reference the [usage in ckb](https://github.com/nervosnetwork/ckb#how-to-contribute) .

File: devtools/_extension-templete/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+[package]
+name = "extension-templete"
+version = "0.1.0"
```
Comment (rev-chaos) [inline devtools/_extension-templete/Cargo.toml:3]:
```suggestion
version = "0.2.0-beta.4"
```

**Review by rev-chaos — APPROVED** [review]

**Review by EthanYuan — APPROVED** [review]

**Pull request merged**